src/compiler/core.ts

```diff
@@ -835,7 +835,7 @@ export function sortAndDeduplicate<T>(array: readonly string[]): SortedReadonlyA
 export function sortAndDeduplicate<T>(array: readonly T[], comparer: Comparer<T>, equalityComparer?: EqualityComparer<T>): SortedReadonlyArray<T>;
 /** @internal */
 export function sortAndDeduplicate<T>(array: readonly T[], comparer?: Comparer<T>, equalityComparer?: EqualityComparer<T>): SortedReadonlyArray<T> {
-    return deduplicateSorted(sort(array, comparer), equalityComparer || comparer || compareStringsCaseSensitive as any as Comparer<T>);
+    return deduplicateSorted(toSorted(array, comparer), equalityComparer || comparer || compareStringsCaseSensitive as any as Comparer<T>);

```

Comment:
I have always wondered, given `deduplicateSorted` used only by `sortAndDeduplicate` can we mark it to make readyOnly to ReadWrite array when length is not zero and work on that array itself instead of creating new array.